app/src/main/java/org/blitzortung/android/app/BackgroundModeHandler.kt

```diff
@@ -0,0 +1,54 @@
+package org.blitzortung.android.app
+
+import android.content.Context
+import android.os.Handler
+import android.util.Log
+import org.blitzortung.android.app.event.BackgroundModeEvent
+import org.blitzortung.android.protocol.ConsumerContainer
+
+class BackgroundModeHandler(private val context: Context)  {
+    val handler = Handler(context.mainLooper)
+
+    private var runable: Runnable? = null
+
+    private val consumerContainer = object : ConsumerContainer<BackgroundModeEvent>() {
+        override fun addedFirstConsumer() { }
+
+        override fun removedLastConsumer() { }
+    }
+
+    fun updateBackgroundMode(isInBackground: Boolean) {
+
+        //Post the isInBackground = true after 500ms
+        //When the users switches activities, the flag is set to true and shortly after to false,
+        //so we shouldn't send out a broadcast
+        if(isInBackground) {
+            runable = Runnable {
+                sendUpdates(isInBackground)
+
+                runable = null
+            }
+
+            handler.postDelayed(runable, 500)
+        } else {
+            if(runable is Runnable) {
+                handler.removeCallbacks(runable)
```

Comment:
When the runnable is executed, it sets the field runnable to null. This is not the case when the runnable is removed before it is executed. In this case the field runnable is still set.